lib/ansible/module_utils/openstack.py

```diff
@@ -114,16 +114,22 @@ def openstack_cloud_from_module(module, min_version='0.12.0'):
         # Due to the name shadowing we should import other way
         import importlib
         sdk = importlib.import_module('openstack')
+        sdk_version = importlib.import_module('openstack.version')
     except ImportError:
         module.fail_json(msg='openstacksdk is required for this module')
 
```

Comment:
> The current code isn't compatible with openstacksdk < 0.10.0.
> Because this is a module_utils, other module not belonging to the ansible repository might use this method with  another value than the default one.

Sounds like min_version should be set to max(StrictVersion('0.10.0'), StrictVersion(min_version)) here.
This would avoid the redundancy below, too.

Reply:
Some modules rely on this default value and don't work with `0.10.0`.

For example using `os_security_group`:
1. with `devel` branch
2. the following diff applied
    ```
    -def openstack_cloud_from_module(module, min_version='0.12.0'):
    +def openstack_cloud_from_module(module, min_version='0.10.0'):
    ```
3. `openstacksdk==0.10.0` installed (same error occurs with `0.11.0`)
4. this task:
    ```
    - os_security_group:
        cloud: mycloud
        state: present
        name: foo
    ```

this exception occurs:
```
TASK [os_security_group] ***
An exception occurred during task execution. To see the full traceback, use -vvv.
The error was: AttributeError: 'Connection' object has no attribute 'current_project_id'
fatal: [localhost]: FAILED! => {
    "changed": false, 
    "rc": 1
}

MSG:

MODULE FAILURE
See stdout/stderr for the exact error


MODULE_STDERR:

Traceback (most recent call last):
  File "$HOME/.ansible/tmp/ansible-tmp-1556617879.09-260636442237090/AnsiballZ_os_security_group.py", line 125, in <module>
    _ansiballz_main()
  File "$HOME/.ansible/tmp/ansible-tmp-1556617879.09-260636442237090/AnsiballZ_os_security_group.py", line 117, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "$HOME/.ansible/tmp/ansible-tmp-1556617879.09-260636442237090/AnsiballZ_os_security_group.py", line 54, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File "/tmp/ansible_os_security_group_payload_uLUg4r/__main__.py", line 163, in <module>
  File "/tmp/ansible_os_security_group_payload_uLUg4r/__main__.py", line 123, in main
AttributeError: 'Connection' object has no attribute 'current_project_id'
```

Then if `0.10.0` is used by default, callers must be modified in order to pass `0.12.0` when required. Is that OK with you ?

Reply:
> Some modules rely on this default value and don't work with 0.10.0.

No doubts about this.

> Then if 0.10.0 is used by default, callers must be modified in order to pass 0.12.0 when required. Is that OK with you ?

Sorry, I did not mean to use 0.10.0 as the default, but process the bigger value, either 0.10.0 or the parmeter value, which is 0.12.0 by default.
This way we would use 0.10.0 if < 0.10.0 is requested or 0.12.0 if the default parmater value is used. If the parameter is set explicitly to > 0.10.0, we would use this.
Is this explained understandable?

Reply:
Yes, thanks!

PR updated.